packages/mdc-dialog/index.js

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+
+import {MDCComponent} from '@material/base';
+import MDCDialogFoundation from './foundation';
+import * as util from './util';
+
+export {MDCDialogFoundation};
+
+export class MDCDialog extends MDCComponent {
+  static attachTo(root) {
+    return new MDCDialog(root);
+  }
+
+  get open() {
+    return this.foundation_.isOpen();
+  }
+
+  set open(value) {
+    if (value) {
+      this.foundation_.open();
+    } else {
+      this.foundation_.close();
+    }
+  }
+
+  get acceptButton() {
+    return this.root_.querySelector(MDCDialogFoundation.strings.ACCEPT_SELECTOR);
+  }
+
+  get cancelButton() {
+    return this.root_.querySelector(MDCDialogFoundation.strings.CANCEL_SELECTOR);
+  }
+
+  get dialog() {
+    return this.root_.querySelector(MDCDialogFoundation.strings.DIALOG_SURFACE_SELECTOR);
+  }
+
+  getDefaultFoundation() {
+    const {FOCUSABLE_ELEMENTS, SCROLL_LOCK_TARGET} = MDCDialogFoundation.strings;
+
+    return new MDCDialogFoundation({
+      hasClass: (className) => this.root_.classList.contains(className),
+      addClass: (className) => this.root_.classList.add(className),
+      removeClass: (className) => this.root_.classList.remove(className),
+      addScrollLockClass: () =>
+        document.querySelector(SCROLL_LOCK_TARGET).classList.add(MDCDialogFoundation.cssClasses.SCROLL_LOCK),
+      removeScrollLockClass: () =>
+        document.querySelector(SCROLL_LOCK_TARGET).classList.remove(MDCDialogFoundation.cssClasses.SCROLL_LOCK),
+      registerInteractionHandler: (evt, handler) =>
+        this.root_.addEventListener(evt, handler, util.applyPassive()),
+      deregisterInteractionHandler: (evt, handler) =>
+        this.root_.removeEventListener(evt, handler, util.applyPassive()),
+      registerDialogInteractionHandler: (evt, handler) =>
+        this.dialog.addEventListener(evt, handler),
+      deregisterDialogInteractionHandler: (evt, handler) =>
+        this.dialog.removeEventListener(evt, handler),
+      registerDocumentKeydownHandler: (handler) => document.addEventListener('keydown', handler),
+      deregisterDocumentKeydownHandler: (handler) => document.removeEventListener('keydown', handler),
+      registerAcceptHandler: (handler) => this.acceptButton.addEventListener('click', handler),
+      deregisterAcceptHandler: (handler) => this.acceptButton.removeEventListener('click', handler),
+      registerCancelHandler: (handler) => this.cancelButton.addEventListener('click', handler),
+      deregisterCancelHandler: (handler) => this.cancelButton.removeEventListener('click', handler),
+      registerFocusTrappingHandler: (handler) => document.addEventListener('focus', handler, true),
+      deregisterFocusTrappingHandler: (handler) => document.removeEventListener('focus', handler, true),
+      numFocusableElements: () => this.dialog.querySelectorAll(FOCUSABLE_ELEMENTS).length,
+      resetDialogFocus: () => this.dialog.querySelectorAll(FOCUSABLE_ELEMENTS)[0].focus(),
```

Comment:
I feel like there's some implicit knowledge captured here that's not made apparent by the method name. How do I know that `resetDialogFocus` is supposed to focus the first focusable element within the dialog?